diff --git a/Complete.pm b/Complete.pm
index 601e495..514daa8 100644
--- a/Complete.pm
+++ b/Complete.pm
@@ -66,9 +66,10 @@ Wayne Thompson
 
 =cut
 
-our($complete, $kill, $erase1, $erase2, $tty_raw_noecho, $tty_restore, $stty, $tty_safe_restore);
+our($die, $complete, $kill, $erase1, $erase2, $tty_raw_noecho, $tty_restore, $stty, $tty_safe_restore);
 our($tty_saved_state) = '';
 CONFIG: {
+    $die      = "\003";
     $complete = "\004";
     $kill     = "\025";
     $erase1 =   "\177";
@@ -144,7 +145,7 @@ sub Complete {
                 $_ eq $kill && do {
                     if ($r) {
                         $r	= 0;
-			$return	= "";
+                        $return	= "";
                         print("\r\n");
                         redo LOOP;
                     }
@@ -168,6 +169,12 @@ sub Complete {
                     print;
                     last CASE;
                 };
+                
+                # (^C) die and close all
+                $_ eq $die && do {
+                    system $tty_restore;
+                    die "\r\n";
+                };
             }
         }
     }
